mapping/document.go

```diff
@@ -525,19 +525,25 @@ func (dm *DocumentMapping) processProperty(property interface{}, path []string,
 		if !propertyValue.IsNil() {
 			switch property := property.(type) {
 			case encoding.TextMarshaler:
-
-				txt, err := property.MarshalText()
-				if err == nil && subDocMapping != nil {
-					// index by explicit mapping
+				// ONLY process TextMarshaler if there is an explicit mapping
+				// AND all of the fiels are of type text
+				// OTHERWISE process field without TextMarshaler
+				if subDocMapping != nil {
+					allFieldsText := true
 					for _, fieldMapping := range subDocMapping.Fields {
-						if fieldMapping.Type == "text" {
+						if fieldMapping.Type != "text" {
+							allFieldsText = false
+						}
+					}
+					txt, err := property.MarshalText()
+					if err == nil && allFieldsText {
```

Comment:
another non-important thought (as it was in the original code, too)... optimization where conversion to string(txt) can happen just once outside the loop

Reply:
done